pkg/gofr/datasource/file/s3/file.go

```diff
@@ -0,0 +1,475 @@
+package s3
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+type s3file struct {
+	conn         *s3.Client
+	name         string
+	offset       int64
+	logger       Logger
+	metrics      Metrics
+	size         int64
+	contentType  string
+	body         io.ReadCloser
+	lastModified time.Time
+}
+
+const (
+	statusErr     = "ERROR"
+	statusSuccess = "SUCCESS"
+)
+
+// Name returns the base name of the file.
+//
+// For a file, this method returns the name of the file without any directory components.
+// For directories, it returns the name of the directory.
+func (f *s3file) Name() string {
+	bucketName := strings.Split(f.name, string(filepath.Separator))[0]
```

Comment:
I'm sure there is a filepath method for this

This comment is valid for each method

Reply:
I wasn't able to find one that can give me only the very root element.